misc/atrium-specs/src/commonMain/kotlin/ch/tutteli/atrium/specs/integration/AbstractCollectionExpectationsTest.kt

```diff
@@ -0,0 +1,85 @@
+package ch.tutteli.atrium.specs.integration
+
+import ch.tutteli.atrium.api.fluent.en_GB.*
+import ch.tutteli.atrium.creating.Expect
+import ch.tutteli.atrium.specs.*
+import ch.tutteli.atrium.specs.integration.utils.ExpectationCreatorTestData
+import ch.tutteli.atrium.specs.integration.utils.SubjectLessTestData
+import ch.tutteli.atrium.testfactories.TestFactory
+import ch.tutteli.atrium.translations.DescriptionCollectionExpectation
+
+@Suppress("FunctionName")
+abstract class AbstractCollectionExpectationsTest(
+    private val toBeEmptySpec: Fun0<Collection<Int>>,
+    private val notToBeEmptySpec: Fun0<Collection<Int>>,
+    private val sizeFeatureSpec: Feature0<Collection<Int>, Int>,
+    private val sizeSpec: Fun1<Collection<Int>, Expect<Int>.() -> Unit>
+) : ExpectationFunctionBaseTest() {
+
+
+    @TestFactory
+    fun subjectLessTest() = subjectLessTestFactory(
+        SubjectLessTestData(
+            toBeEmptySpec.forSubjectLessTest(),
+            notToBeEmptySpec.forSubjectLessTest(),
+            sizeFeatureSpec.forSubjectLessTest().withFeatureSuffix(),
+            sizeSpec.forSubjectLessTest { toBeLessThan(2) })
+    )
```

Comment:
nitpick, you don't need SubjectLessTestData if you have only one, you could write:
```suggestion
    fun subjectLessTest() = subjectLessTestFactory(
        toBeEmptySpec.forSubjectLessTest(),
        notToBeEmptySpec.forSubjectLessTest(),
        sizeFeatureSpec.forSubjectLessTest().withFeatureSuffix(),
        sizeSpec.forSubjectLessTest { toBeLessThan(2) }
    )
```

Reply:
@robstoll - Done.